src/ipnet.rs

```diff
@@ -606,6 +627,24 @@ impl Ipv6Net {
         self.addr.is_multicast()
     }
 
+    /// Return `true` if this address is a multicast address with
+    /// administrative scope (site-local or organization-local) as defined in
+    /// [RFC 4291].
+    ///
+    /// [RFC 4291]: https://tools.ietf.org/html/rfc4291
+    pub const fn is_admin_scoped_multicast(&self) -> bool {
+        if !self.addr.is_multicast() {
+            return false;
+        }
+
+        // Extract the scope field (bits 4-7 of the second byte)
+        let segments = self.addr.segments();
+        let scope = (segments[0] & 0x000F) as u8;
+
+        // RFC 4291: Scope values 5 (site-local) and 8 (organization-local)
+        matches!(scope, 0x5 | 0x8)
```

Comment:
I *think* `0x4` should be included here, at least going off [RFC 7346](https://www.rfc-editor.org/rfc/rfc7346.html):

> Interface-Local, Link-Local, and Realm-Local scope boundaries are
      automatically derived from physical connectivity or other non-
      multicast-related configurations.  Global scope has no boundary.
      The boundaries of all other non-reserved scopes of Admin-Local or
      larger are administratively configured.

Reply:
added